pkg/models/repository_configuration.go

```diff
@@ -48,12 +51,28 @@ func (rc *RepositoryConfiguration) BeforeCreate(tx *gorm.DB) (err error) {
 
 // BeforeUpdate perform validations of Repository Configurations
 func (rc *RepositoryConfiguration) BeforeUpdate(tx *gorm.DB) (err error) {
+	if err := rc.UniqueVersions(tx); err != nil {
+		return err
+	}
 	if err := rc.validate(); err != nil {
 		return err
 	}
 	return nil
 }
 
+func (rc *RepositoryConfiguration) UniqueVersions(tx *gorm.DB) (err error) {
```

Comment:
nitpick, but I like the name `DedupeVersions` (or maybe `UniqifyVersions`) because deduplication is the action being performed on the versions. I feel it communicates the intention of the function more clearly.

Reply:
oh yeah i like dedup.  Uniqify was the best i came up with and i'm not sure thats actually a word hah.

Although, should it be Dedup or Dedupe  ?

Reply:
I'm not sure :). I don't think either of them are an actual word. "Deduplicate" would be proper. I went with dedupe because it's spelled how it sounds.

Reply:
went with Dedupe